stdlib/public/core/ExistentialCollection.swift

```diff
@@ -1303,10 +1303,26 @@ extension AnySequence {
   }
 
   @inlinable
-  public func map<T>(
+  @_alwaysEmitIntoClient
+  public func map<T, E>(
+    _ transform: (Element) throws(E) -> T
+  ) throws(E) -> [T] {
+    do {
+      return try _box._map(transform)
```

Comment:
Would there be an advantage to adding a typed-error version of `ExistentialCollection._map()` ?

Reply:
I don't think we can, because it's a fixed-layout type

Reply:
According to `UnderscoredAttributes.md` (if it's correct), new virtual methods can be added. It would avoid the catch-and-downcast dance, and if that happened to generate better code it might be worth it.

Reply:
These methods are `@inlineable`, which implies `@usableFromInline`, so they can't go away. Additionally, their callers need to be `@_alwaysEmitIntoClient`, so we couldn't call into different or new entry points for a typed map without having a dynamic `if #available` check, which seems worse than this forced cast.

I think what we have is the best we're going to get.